pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -717,3 +729,18 @@ func WaitForBuildComplete(c osclient.BuildInterface, name string) error {
 		}
 	}
 }
+
+func isInvalidSourceInputsError(err error) bool {
+	if err != nil {
+		if statusErr, ok := err.(*kerrors.StatusError); ok {
+			if statusErr.ErrStatus.Reason == "Invalid" {
+				for _, cause := range statusErr.ErrStatus.Details.Causes {
```

Comment:
No, just use the source field "spec.source" instead of the message (you get that separate from message).  And don't check Reason against string "Invalid", check the error type against `api/errors.IsInvalid`